pkg/asset/machines/gcp/zones.go

```diff
@@ -48,3 +49,39 @@ func AvailabilityZones(project, region string) ([]string, error) {
 	sort.Strings(zones)
 	return zones, nil
 }
+
+// ZonesForInstanceType retrieves a filtered list of availability zones where
+// the particular instance type is available. This is mainly necessary for
+// arm64, since the instance t2a-standard-* is not available in all
+// availability zones.
+func ZonesForInstanceType(project, region, instanceType string) ([]string, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+	defer cancel()
+
+	ssn, err := gcpconfig.GetSession(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get session")
+	}
+
+	svc, err := compute.NewService(ctx, option.WithCredentials(ssn.Credentials))
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create compute service")
+	}
+
+	found := sets.New[string]()
+	filter := fmt.Sprintf("name = \"%s\" AND zone : %s-*", instanceType, region)
```

Comment:
can we get this routine in a common place and ref that function here?

Reply:
Addressed the the latest update https://github.com/openshift/installer/pull/7096/commits/c7e91d546f4c963fe8efe82ef585544a84110016